### ISSUE TITLE
GCC port doesn't run - PT_INTERP ELF phdr in statically compiled gcc binary

### ISSUE THREAD

**Issue by shawnanastasio** [SerenityOS/serenity#4697]
Serenity version: 3d02597316f60fe017582ac32c19ff7844f92eb7

Building gcc from Ports/ and trying to run it in serenity results in the following being printed to the serial console:
```
Found PT_INTERP header (1) in non-DYN ELF object! What? We can't handle this!
[Shell(27:27)]: exec(/usr/local/bin/gcc): File has invalid ELF Program headers
Shell(27): execvpe() failed on attempt (/usr/local/bin/gcc) with Exec format error
Shell(27): execvp() about to return -1 with errno=8
```

After some inspection with readelf, I see that the `gcc` binary has an `INTERP` program header despite having a type of `EXEC`:
```
# readelf -l /usr/local/bin/gcc

Elf file type is EXEC (Executable file)
Entry point 0x804de91
There are 5 program headers, starting at offset 52

Program Headers:
  Type           Offset   VirtAddr   PhysAddr   FileSiz MemSiz  Flg Align
  PHDR           0x34 0x8048034 0x8048034 0xa0 0xa0 R   0x4
  INTERP         0xd4 0x80480d4 0x80480d4 0x13 0x13 R   0x1
      [Requesting program interpreter: /usr/lib/Loader.so]
  LOAD           0x0 0x8048000 0x8048000 0xf62e8 0xf62e8 R E 0x1000
  LOAD           0xf7000 0x813f000 0x813f000 0x2254 0x5fb8 RW  0x1000
  DYNAMIC        0xf8838 0x8140838 0x8140838 0xd0 0xd0 RW  0x4

 Section to Segment mapping:
  Segment Sections...
    0     
    1     .interp 
    2     .interp .hash .dynsym .dynstr .rel.dyn .rel.plt .init .plt .plt.got .text .fini .rodata .eh_frame .gcc_except_table 
    3     .init_array .ctors .dtors .data.rel.ro .dynamic .got .got.plt .data .bss 
```

Maybe someone more familiar with the toolchain and newly added dynamic linker can chime in? It looks like GCC is being completely miscompiled here.

**Comment by shawnanastasio** [comment]
Actually, looking at my system's native `gcc` binary, it is also type `EXEC` with an `INTERP` header. Maybe the loader should just treat it as if it was a `DYN` then? I'll have to look at the ELF spec to see what behavior is expected here...

**Comment by shawnanastasio** [comment]
My bad - looks like this was fixed by #4503. I thought I updated my local checkout before trying but I guess not. Sorry again!
